### PR TITLE
Add Ethereum precompiles

### PR DESCRIPTION
**Pull request by AsceticBear**
Ethereum precompile list:
https://github.com/ethereum/go-ethereum/blob/bd6879ac518431174a490ba42f7e6e822dcb3ee1/core/vm/contracts.go#L51-L60

### CODE REVIEW

File: node/runtime/pangolin/src/pallets/evm_.rs [hunk 1]
```diff
@@ -133,13 +138,17 @@ where
...
+			// Ethereum precompiles:
 			a if a == addr(1) => Some(ECRecover::execute(input, target_gas, context, is_static)),
 			a if a == addr(2) => Some(Sha256::execute(input, target_gas, context, is_static)),
 			a if a == addr(3) => Some(Ripemd160::execute(input, target_gas, context, is_static)),
 			a if a == addr(4) => Some(Identity::execute(input, target_gas, context, is_static)),
+			a if a == addr(5) => Some(Modexp::execute(input, target_gas, context, is_static)),
+			a if a == addr(6) => Some(Bn128Add::execute(input, target_gas, context, is_static)),
+			a if a == addr(7) => Some(Bn128Mul::execute(input, target_gas, context, is_static)),
+			a if a == addr(8) => Some(Bn128Pairing::execute(input, target_gas, context, is_static)),
 			a if a == addr(9) => Some(Blake2F::execute(input, target_gas, context, is_static)),
```
Comment (HackFisher) [inline node/runtime/pangolin/src/pallets/evm_.rs:150]: For address 9, is it from ethereum or upstream frontier?

File: node/runtime/pangolin/src/pallets/evm_.rs [hunk 2]
```diff
@@ -133,13 +138,17 @@ where
 		};
 
 		match address {
-			// Ethereum precompiles
+			// Ethereum precompiles:
 			a if a == addr(1) => Some(ECRecover::execute(input, target_gas, context, is_static)),
 			a if a == addr(2) => Some(Sha256::execute(input, target_gas, context, is_static)),
 			a if a == addr(3) => Some(Ripemd160::execute(input, target_gas, context, is_static)),
 			a if a == addr(4) => Some(Identity::execute(input, target_gas, context, is_static)),
```
Comment (HackFisher) [inline node/runtime/pangolin/src/pallets/evm_.rs:145]: Is this same with Ethereum's dataCopy?

```
common.BytesToAddress([]byte{4}): &dataCopy{},
```

File: node/runtime/pangolin/src/pallets/evm_.rs (hunk 1 above)
Comment (AsceticBear) [reply node/runtime/pangolin/src/pallets/evm_.rs:150]: Upstream Frontier

Comment (HackFisher) [reply node/runtime/pangolin/src/pallets/evm_.rs:150]: https://ethereum.stackexchange.com/questions/15479/list-of-pre-compiled-contracts

And also checked with moonbeam too.

**Review by HackFisher — APPROVED** [review]

**Pull request merged**